### PR TITLE
Final review before production

### PR DESCRIPTION
**Pull request by Metsanpeitto**
This is the last merge that will occur before production.
All the tasks have been properly divided into Kanban cards, and those Kanban cards were translated into branches that were submitted to review by other team members.
After the approval of each pull request, those branches were merged into developing, which now holds a totally functional version of the app.

The front end pull request link is this:
[Frontend Pull Request](https://github.com/Metsanpeitto/microverse_final_project_front/pull/27/)

The app has been already deployed on Heroku 
[API Deployed](https://frozen-dusk-66130.herokuapp.com/api-docs/index.html)

### CODE REVIEW

File: app/controllers/current_user_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class CurrentUserController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    render json: current_user, status: :ok
```
Comment (rvvergara) [inline app/controllers/current_user_controller.rb:4]: Hi guys. Im curious as to how exactly we're using this and how necessary this method is.

Our API should be stateless, meaning it shouldn't be the one keeping state of a user's authenticated condition. Therefore, having a "current_user" in the backend might not seem intuitive.

File: app/controllers/reservations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  before_action :authenticate_user!, only: %i[create update]
+
+  def index
+    @reservations = Reservation.all
+    render json: @reservations
+  end
+
+  # GET /reservations/1
+  def show
+    render json: @reservation
```
Comment (rvvergara) [inline app/controllers/reservations_controller.rb:12]: don't you think we should also require only authenticated users to access this endpoint?

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Rails.application.routes.draw do
+  mount Rswag::Ui::Engine => '/api-docs'
+  mount Rswag::Api::Engine => '/api-docs'
+  root to: "items#index"
+  get 'current_user/index'
```
Comment (rvvergara) [inline config/routes.rb:5]: Ditto in relation to my comment in the current_user controller. How is this endpoint necessary for the app?

**Review by rvvergara — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

Two major things I notice:

- I am not sure if this review request is only meant for your back end but I didn't seem to see any link to your front end app for me to review. Please make sure you include a link to it in your succeeding code review request.

- We need to write tests for our API


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/controllers/current_user_controller.rb (hunk 1 above)
Comment (Metsanpeitto) [reply app/controllers/current_user_controller.rb:4]: :heavy_check_mark: Removed! 
It was not needed, indeed. :+1:

File: app/controllers/reservations_controller.rb (hunk 2 above)
Comment (Metsanpeitto) [reply app/controllers/reservations_controller.rb:12]: :heavy_check_mark: Updated! 
```cmd
  before_action :authenticate_user!, only: %i[create update show]
```
:+1:

File: config/routes.rb (hunk 3 above)
Comment (Metsanpeitto) [reply config/routes.rb:5]: :heavy_check_mark: Removed! 
It was not needed, indeed. :+1:

**Comment by Metsanpeitto** [comment]
@rvvergara  Hola!
Thanks for your time and for pointing out those things. I fixed them already.

However. In my pull request : 
```cmd 
This is the last merge that will occur before production.
All the tasks have been properly divided into Kanban cards, and those Kanban cards were translated into branches that were submitted to review by other team members.
After the approval of each pull request, those branches were merged into developing, which now holds a totally functional version of the app.

The front end pull request link is this:
Frontend Pull Request    <-----    THIS LINK IS THE FRONT END PULL REQUEST 
```
Please visit the link here: 

The front end pull request link is this:
[Frontend Pull Request](https://github.com/Metsanpeitto/microverse_final_project_front/pull/27/)

We have also implemented RSwag documentation that requires RSpec in order to work successfully.
Visit the API documentation in the deployed version : 

### API Documentation

[API Documentation](https://frozen-dusk-66130.herokuapp.com/api-docs/index.html)



The app has been already deployed on Heroku 
[API Deployed](https://frozen-dusk-66130.herokuapp.com/api-docs/index.html)

File: app/controllers/current_user_controller.rb (hunk 1 above)
Comment (rvvergara) [reply app/controllers/current_user_controller.rb]: Great!!

File: app/controllers/reservations_controller.rb (hunk 2 above)
Comment (rvvergara) [reply app/controllers/reservations_controller.rb:12]: Awesome!

**Review by rvvergara — APPROVED** [review]
Hi team, 

In my opinion, your back end is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**